docs/HowToGuides/FAQ.md

```diff
@@ -0,0 +1,131 @@
+# Frequently Asked Questions
+
+## Build System and CMake Configuration
+
+### How do I add a new file to CMake?
+
+If you forget to add a new file to the CMake configuration, you may end up with
+undefined symbol errors at link time.
+
+There should be a `CMakeLists.txt` in the directory where you added the new
+file, which has the list of different .h/.cpp files to be included in the build.
+Add your new file to that list.
+
+### How do I speed up iterating on changes to the build system?
+
+The general idea is to build as little as you can.
+- [Use `sccache`](/docs/HowToGuides/GettingStarted.md) or `ccache` if you
+  aren't doing so already.
+- Use `build-script`'s various `--skip-*` flags to skip configuring for
+  platforms that you do not care about.
+- If you're on macOS, use `--swift-darwin-supported-archs="x86_64"`.
+- Use a release build without assertions (`--release --no-assertions`).
+  While debuginfo and assertions are valuable to enable when working on the
+  toolchain itself, they are not so useful if you are working only on changes
+  to the build system.
+
+## Using a Locally Built Toolchain
+
+### How do I use a locally built compiler to build X?
+
+You can use the `SWIFT_EXEC` environment variable to use a locally
+built compiler to compile both packages and Xcode projects.
+
+1. For SwiftPM packages, pass the environment variable when invoking SwiftPM.
+   ```sh
+   # Assuming the current working directory contains the package, build the
+   # package using a custom compiler.
+   SWIFT_EXEC=/path/to/swiftc swift build
+   ```
+2. For Xcode projects, select the project in the Project Navigator. In the
+   Build Settings tab, click '+' and then 'Add User-Defined Setting'.
+   Create a build setting `SWIFT_EXEC` with the value set to `/path/to/swiftc`.
+   If you now do a clean build, your locally built compiler will be used.
+
+   At the time of writing, in the latest Xcode 12 beta, `SWIFT_EXEC` does not
+   work for SwiftPM integration inside Xcode, so this will not work for Xcode
+   projects that depend on SwiftPM packages.
+
+**Note:** Even thought the variable says 'SWIFT', it needs to point to
+'swift**c**', not 'swift'. The extra 'c' is not a typo.
+
+## Testing and CI
+
+### How do I reproduce/fix a test that fails in CI but passes for me locally?
+
+TODO: Write some tips here, point to Testing.md for simulator setup.
+
+## Documentation
+
+### Where can I find documentation on X?
+
+This very depends on what X is, but some broad guidelines are:
+
+1. Do a case-insensitive recursive string search.
+   - Use a specialized tool like [ripgrep](https://github.com/BurntSushi/ripgrep)
+     or [ag](https://github.com/ggreer/the_silver_searcher).
+   - Use 'Find in Workspace' in Xcode (<kbd>⌘</kbd>+<kbd>⇧</kbd>+<kbd>F</kbd>).
+   - Use `grep -i -r "mypattern" .`.
```

Comment:
not sure if it's worth mentioning, but `git grep` is much faster than basic `grep` in my experience, usually on par with `rg`. Additionally, `git grep` has some nice features that none of the other grep tools have.

1. `git grep [-p | --show-function]` – this prints a line before a match showing which function the match is in
2. `git grep [-W | --function-context]` – this shows the whole function for a given match